chain/chain/src/stateless_validation/chunk_validation.rs

```diff
@@ -369,11 +372,18 @@ pub fn pre_validate_chunk_state_witness(
 
     let current_protocol_version =
         epoch_manager.get_epoch_protocol_version(&state_witness.epoch_id)?;
-    if !checked_feature!(
+    let transaction_validity = if checked_feature!(
         "protocol_feature_relaxed_chunk_validation",
         RelaxedChunkValidation,
         current_protocol_version
     ) {
+        let mut check = chain.transaction_validity_check(last_chunk_block.header().clone());
+        state_witness
+            .transactions
+            .iter()
+            .map(|t| check(t))
+            .collect::<Vec<_>>()
```

Comment:
I think this should be the previous block.
AFAIU when checking validity for transactions included in chunk in block X, the validity check expects to receive the block before X.
Chunk witness proves the execution of chunk included in `last_chunk_block`, `witness.chunk_header` is the next chunk that will be applied in the future.